e2e_test/streaming/bug_fixes/stack_overflow_17342.slt

```diff
@@ -0,0 +1,120 @@
+statement ok
+SET streaming_parallelism TO 1;
+
+statement ok
+CREATE TABLE t (v int);
+
+# This query used to overflow the stack during optimization as it generated a left-deep tree
+# of `OR xx IS NOT NULL` expression in the filter after each full outer join.
+skipif madsim
```

Comment:
👀

Reply:
I mean why it doesn't work in madsim?

Reply:
I forgot, but

https://github.com/risingwavelabs/risingwave/blob/14b2f9c76ad20b7c644dfbf118f35205b9fe089d/src/common/src/util/recursive.rs#L98-L102

Reply:
```rust
unsafe fn guess_os_stack_limit() -> Option<usize> {
    let mut attr = std::mem::MaybeUninit::<libc::pthread_attr_t>::uninit();
    assert_eq!(libc::pthread_attr_init(attr.as_mut_ptr()), 0);
    assert_eq!(libc::pthread_getattr_np(libc::pthread_self(),
                                        attr.as_mut_ptr()), 0);
    let mut stackaddr = std::ptr::null_mut();
    let mut stacksize = 0;
    assert_eq!(libc::pthread_attr_getstack(
        attr.as_ptr(), &mut stackaddr, &mut stacksize
    ), 0);
    assert_eq!(libc::pthread_attr_destroy(attr.as_mut_ptr()), 0);
    Some(stackaddr as usize)
}
```

Some assertion failed. Possibly due to libc overridden. Haven't dived into it since it's not a big deal.